privacyidea/lib/log.py

```diff
@@ -141,12 +141,18 @@ def __call__(self, func):
         def log_wrapper(*args, **kwds):
             """
             Wrap the function in log entries. The entry of the function and
-            the exit of the function is logged.
+            the exit of the function is logged using the DEBUG log level.
+            If the logger does not log DEBUG messages, this just returns
+            the result of ``func(*args, **kwds)`` to improve performance.
 
             :param args: The positional arguments starting with index
             :param kwds: The keyword arguemnts
             :return: The wrapped function
             """
+            # Exit early if self.logger disregards DEBUG messages.
+            if not self.logger.isEnabledFor(logging.DEBUG):
```

Comment:
Sure! My line of thinking was: ``log_wrapper`` uses ``self.logger.debug(...)`` to log function entry and exit points. Thus, it may make the effort to hide passwords from ``args`` and ``kwargs`` even if the resulting message logged using ``self.logger.debug`` will never make it to the log (because, e.g., ``PI_LOGLEVEL`` is ``logging.INFO``). So with this change ``log_wrapper`` should exit early if ``self.logger`` does not even forward DEBUG messages to the log.

But I'm not sure whether this is the right way to do it, and I'm not sure if it results in a noticeable speedup :)